lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -1114,6 +1114,7 @@ def install_puppet_agent_pe_promoted_repo_on( hosts, opts )
               if arch == 'x86_64'
                 arch = 'amd64'
               end
+              version = version[0,2] + '.' + version[2,2] if variant =~ /ubuntu/
```

Comment:
The issue here is that 'ubuntu-14.04-amd64' is a valid platform (and would be the platform generated by genconfig) - and then this code would not be necessary.

I'm poking around to see if there is a more elegant solution to this.